pallets/nfts/src/features/approvals.rs

```diff
@@ -107,6 +107,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	) -> DispatchResult {
 		let mut details = Item::<T, I>::get(collection, item).ok_or(Error::<T, I>::UnknownItem)?;
 
+		ensure!(
+			!CollectionApprovals::<T, I>::contains_key((collection, &details.owner, &delegate)),
```

Comment:
Explanation: Throw if there is an existing collection approval for the `delegate`. Can't cancel approval of a specified collection item if the permission to transfer all items within the collection for that `delegate` is enabled.

Reply:
Can we add a comment here as to why cancelling an item approval when the collection is approved is invalid please? This will help reviewers and auditors.